src/main/java/de/tum/in/www1/artemis/web/rest/LectureResource.java

```diff
@@ -383,4 +386,17 @@ public ResponseEntity<Void> deleteLecture(@PathVariable Long lectureId) {
         lectureService.delete(lecture);
         return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, lectureId.toString())).build();
     }
+
+    /**
+     * This method calls the executeLectureIngestionPipeline function with the needed parameters in the right format
+     *
+     * @param lecture the lecture to be updated
+     */
+    private void helpExecuteIngestionPipeline(Lecture lecture) {
+        pyrisWebhookService.ifPresent(service -> {
+            List<AttachmentUnit> attachmentUnitList = lecture.getLectureUnits().stream().filter(lectureUnit -> lectureUnit.getType().equals("attachment"))
+                    .map(lectureUnit -> (AttachmentUnit) lectureUnit).collect(Collectors.toCollection(ArrayList::new));
```

Comment:
This probably does not need to be mutable
```suggestion
                    .map(lectureUnit -> (AttachmentUnit) lectureUnit).toList();
```